app/controllers/home_page_controller.rb

```diff
@@ -4,4 +4,14 @@ class HomePageController < ApplicationController
   def index
     @products = Product.paginate(page: params[:page], per_page: 12)
   end
+
+  private
+
+  def redirect
+    if current_user&.admin? && params[:preview].nil?
+      redirect_to admin_path
+    else
+      true
+    end
+  end
```

Comment:
we usually use another tab indentation for private methods. so this will be the structure
```ruby
class A
  def a
  end

  private

    def private_a
    end
end
```

Reply:
I used do that, but my ruby formatter kept complaining